File: Java/RomanToInteger.java
public class RomanToInteger {

  public static void main(String args[]) {
    System.out.println("\nExample 1: \nIII = " + romanToInt("III"));
    System.out.println("\nExample 2: \nIV = " + romanToInt("IV"));
    System.out.println("\nExample 3: \nIX = " + romanToInt("IX"));
    System.out.println("\nExample 4: \nLVIII = " + romanToInt("LVIII"));
    System.out.println("\nExample 5: \nDCXXI = " + romanToInt("DCXXI"));
    System.out.println("\nExample 6: \nMCMXCIV = " + romanToInt("MCMXCIV"));
  }

  public static int romanToInt(String s) {
    int sum = 0;
    char[] rom = s.toCharArray();

    for (int i = 0; i < rom.length; i++) {
      if (rom[i] == 'M') {
        sum += 1000;
      } else if (rom[i] == 'D') {
        sum += 500;
      } else if (rom[i] == 'L') {
        sum += 50;
      } else if (rom[i] == 'V') {
        sum += 5;
      } else if (rom[i] == 'C') {
        if (i != rom.length - 1) {
          if (rom[i + 1] == 'D') {
            sum += 400;
            i++;
          } else if (rom[i + 1] == 'M') {
            sum += 900;
            i++;
          } else {
            sum += 100;
          }
        } else {
          sum += 100;
        }
      } else if (rom[i] == 'X') {
        if (i != rom.length - 1) {
          if (rom[i + 1] == 'L') {
            sum += 40;
            i++;
          } else if (rom[i + 1] == 'C') {
            sum += 90;
            i++;
          } else {
            sum += 10;
          }
        } else {
          sum += 10;
        }
      } else if (rom[i] == 'I') {
        if (i != rom.length - 1) {
          if (rom[i + 1] == 'V') {
            sum += 4;
            i++;
          } else if (rom[i + 1] == 'X') {
            sum += 9;
            i++;
          } else {
            sum += 1;
          }
        } else {
          sum += 1;
        }
      } else {
        System.out.println("Something went wrong!");
      }
    }
    return sum;
  }
}

File: Python/RomanToInteger.py
def romanToInt(s):
  """Calculates value of Roman numeral string

  Args:
      s (string): String of Roman numeral characters being analyzed

  Returns:
      int: integer value of Roman numeral string
  """
  sum = 0
  i = 0

  while i < len(s):
    if s[i] == 'M':
      sum += 1000
    elif s[i] == 'D':
      sum += 500
    elif s[i] == 'L':
      sum += 50
    elif s[i] == 'V':
      sum += 5
    elif s[i] == 'C':
      if i < len(s) - 1:
        if s[i + 1] == 'M':
          sum += 900
          i += 1
        elif s[i + 1] == 'D':
          sum += 400
          i += 1
        else:
          sum += 100
      else:
        sum += 100
    elif s[i] == 'X':
      if i < len(s) - 1:
        if s[i + 1] == 'C':
          sum += 90
          i += 1
        elif s[i + 1] == 'L':
          sum += 40
          i += 1
        else:
          sum += 10
      else:
        sum += 10
    elif s[i] == 'I':
      if i < len(s) - 1:
        if s[i + 1] == 'X':
          sum += 9
          i += 1
        elif s[i + 1] == 'V':
          sum += 4
          i += 1
        else:
          sum += 1
      else:
        sum += 1
    else:
      return "Something went wrong"
    i += 1

  return sum

def main():
  print(
    format("%-10s %-10s %10s" % ("Test", "String", "Integer")) + "\n" +
    "-" * 32 + "\n" +
    format("%-10s %-10s %10d" % ("Test 1:", "III", romanToInt("III"))) + "\n" +
    format("%-10s %-10s %10d" % ("Test 2:", "IV", romanToInt("IV"))) + "\n" +
    format("%-10s %-10s %10d" % ("Test 3:", "IX", romanToInt("IX"))) + "\n" +
    format("%-10s %-10s %10d" % ("Test 4:", "LVIII", romanToInt("LVIII"))) + "\n" +
    format("%-10s %-10s %10d" % ("Test 5:", "MCMXCIV", romanToInt("MCMXCIV")))
  )

main()